catalog/app/containers/Bucket/File.js

```diff
@@ -453,6 +459,16 @@ export default function File({
               onChange={onViewModeChange}
             />
           )}
+          {!!editorState.type.brace && (
+            <FileEditor.Controls
+              disabled={editorState.saving}
+              editing={editorState.editing}
+              className={classes.button}
+              onSave={editorState.onSave}
+              onCancel={editorState.onCancel}
+              onEdit={editorState.onEdit}
```

Comment:
maybe it would make sense to add a special object holding all these props together for convenience, so we can do smth like `<FileEditor.Controls {...{ editorState.controlsProps }} />`

Reply:
I thought about two options:
1. pass the whole `editorState` to <Controls />
2. pass actions: `<Controls actions={editorState.actions}/>`

But didn't decide yet

Reply:
If you have some comments about using this pattern, I'll address them in the next iterations. It's very similar to using mutable `this` in OOP, so maybe you have some concerns about it.

Reply:
> It's very similar to using mutable this in OOP, so maybe you have some concerns about it.

i don't see a lot of similarities tbh)